src/metacognition/confidence_estimator.py

```
@@ -0,0 +1,466 @@
+    """
+
+    # Weights for each factor
+    FACTOR_WEIGHTS = {
+        ConfidenceFactor.GOAL_CLARITY: 0.25,
+        ConfidenceFactor.MEMORY_SUPPORT: 0.15,
+        ConfidenceFactor.TOOL_AVAILABILITY: 0.15,
+        ConfidenceFactor.CONTEXT_FAMILIARITY: 0.15,
+        ConfidenceFactor.PLAN_COMPLETENESS: 0.10,
+        ConfidenceFactor.SUCCESS_HISTORY: 0.15,
+        ConfidenceFactor.ERROR_STATE: 0.05,
+    }
```

Ruff (0.14.10)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)